src/robotlibcore.py

```diff
@@ -142,6 +135,38 @@ def get_keyword_documentation(self, name):
             doc = '{}\n\n{}'.format(doc, tags) if doc else tags
         return doc
 
+    def get_keyword_types(self, keyword_name):
+        method = self.keywords.get(keyword_name)
+        if keyword_name == '__init__':
+            method = self.__init__
+        elif keyword_name.startswith('__') and keyword_name.endswith('__'):
+            return {}
+        if not method:
+            raise ValueError('%s is not keyword.' % keyword_name)
+        robot_types = getattr(method, 'robot_types', ())
+        if robot_types is None:
+            return robot_types
+        annotations = self._get_annotations(method)
+        types = self._join_defaults_with_annotations(method, annotations)
+        if robot_types:
+            types.update(robot_types)
+        return types
+
+    def _get_annotations(self, method):
+        if PY2:
+            return {}
+        annotations = typing.get_type_hints(method)
```

Comment:
1. `typing.get_type_hints` returns type hints, not anymore original annotations. Thus `type_hints` or `hints` would be a better variable name and probably should be used in the method name as well.

2. `typing.get_type_hints` can blow up with *any* exception. We could handle that like

    ```python
    try:
        hints = typing.get_type_hints(method)
    except Exception:
        hints = method.__annotations__
    ```

   but I'm not sure is it a good idea in this context. Perhaps it would be better to report the error explicitly? A test is needed either way. You can find related Robot side code [here](https://github.com/robotframework/robotframework/blob/master/src/robot/running/arguments/argumentparser.py#L97).

Reply:
Well, perhaps it is better emulate the Robot side here. Went for a similar implementation.